web/components/links/create-links-button.tsx

```diff
@@ -0,0 +1,192 @@
+import clsx from 'clsx'
+import { useState } from 'react'
+import { Col } from '../layout/col'
+import { Row } from '../layout/row'
+import { Title } from '../title'
+import { User } from 'common/user'
+import { ManalinkCardPreview, ManalinkInfo } from 'web/components/manalink-card'
+import { createManalink } from 'web/lib/firebase/manalinks'
+import { Modal } from 'web/components/layout/modal'
+import Textarea from 'react-expanding-textarea'
+import dayjs from 'dayjs'
+import Button, { SubmitButton } from '../button'
+import getLinkUrl from 'functions/src/get-manalink-url'
+import { DuplicateIcon } from '@heroicons/react/outline'
+
+export function CreateLinksButton(props: {
+  user: User
+
+  highlightedSlug: string
+  setHighlightedSlug: any
+}) {
+  const { user, highlightedSlug, setHighlightedSlug } = props
+
+  const [open, setOpen] = useState(false)
+
+  function updateOpen(newOpen: boolean) {
+    setOpen(newOpen)
+  }
+
+  return (
+    <>
+      <Modal open={open} setOpen={updateOpen}>
+        <Col className="gap-4 rounded-md bg-white px-8 py-6">
+          <CreateManalinkForm
+            highlightedSlug={highlightedSlug}
+            user={user}
+            onCreate={async (newManalink) => {
+              const slug = await createManalink({
+                fromId: user.id,
+                amount: newManalink.amount,
+                expiresTime: newManalink.expiresTime,
+                maxUses: newManalink.maxUses,
+                message: newManalink.message,
+              })
+              setHighlightedSlug(slug || '')
```

Comment:
You could extract this one to an `onCreate` function defined above since it is longer.